src/Listener/RestRequestListener.php

```diff
@@ -68,19 +87,21 @@ public function __construct(
     public function onKernelController($event)
     {
         $request = $event->getRequest();
+        // We need make sure firewall allows the client to make the request,
+        // so instead we validate request here
+        if ($this->requestHelper->isRestRequest($request)) {
+            $options = $this->requestHelper->getOptionsFromRequest($request);
+            $this->checkOptions($request, $options);
```

Comment:
Few notes:
- I'd avoid calling `isRestRequst` here, as it does not help understanding what the code is doing. The natural question that could rise here would be "why are we doing something below this `if`, if that't not REST request at all?" I think just using `geOptionsFromRequest` could be used here
- wouldn't it be clearer if we'd "check options" in the bottom of this function at all cases, instead of moving that to a separate method? Or are you avoiding `setOptionsForRequest` if they were already set previously?
- if moving to a private method (which is fine), I'd change the naming – `check` implies that we just validate something. In this case we're propagating all the needed attributes for the request;
- I'd put that private method right below this one (and/or above `checkRequiredPermissions`). This allows better readability – we start with higher scope, then dig deeper and deeper into the details. Currently it's a bit chaotic, as we ding `checkRequiredPermissions` before using it at all